stackit/internal/services/iaas/networkinterface/resource.go

```diff
@@ -0,0 +1,637 @@
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"project_id": schema.StringAttribute{
+				Description: "STACKIT project ID to which the network is associated.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					validate.UUID(),
+					validate.NoSeparator(),
+				},
+			},
+			"network_id": schema.StringAttribute{
+				Description: "The network ID to which the network interface is associated.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					validate.UUID(),
+					validate.NoSeparator(),
+				},
+			},
+			"network_interface_id": schema.StringAttribute{
+				Description: "The network interface ID.",
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+				Validators: []validator.String{
+					validate.UUID(),
+					validate.NoSeparator(),
+				},
+			},
+			"name": schema.StringAttribute{
+				Description: "The name of the network interface.",
+				Optional:    true,
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+					stringvalidator.LengthAtMost(63),
+					stringvalidator.RegexMatches(
+						regexp.MustCompile(`^[A-Za-z0-9]+((-|_|\s|\.)[A-Za-z0-9]+)*$`),
+						"must match expression"),
+				},
+			},
+			"allowed_addresses": schema.ListAttribute{
+				Description: "The list of CIDR (Classless Inter-Domain Routing) notations.",
+				Optional:    true,
+				Computed:    true,
+				ElementType: types.StringType,
+			},
+			"device": schema.StringAttribute{
+				Description: "The device UUID of the network interface.",
+				Computed:    true,
+			},
+			"ipv4": schema.StringAttribute{
+				Description: "The IPv4 address.",
+				Optional:    true,
+				Computed:    true,
+				Validators: []validator.String{
+					validate.IP(),
+				},
+			},
+			"ipv6": schema.StringAttribute{
+				Description: "The IPv6 address.",
+				Optional:    true,
+				Computed:    true,
+				Validators: []validator.String{
+					validate.IP(),
+				},
+			},
+			"labels": schema.MapAttribute{
+				Description: "Labels are key-value string pairs which can be attached to a network interface.",
+				ElementType: types.StringType,
+				Optional:    true,
+			},
+			"mac": schema.StringAttribute{
+				Description: "The MAC address of network interface.",
+				Computed:    true,
+			},
+			"security": schema.BoolAttribute{
+				Description: "The Network Interface Security. If set to false, then no security groups will apply to this network interface.",
+				Computed:    true,
+				Optional:    true,
+			},
+			"security_groups": schema.ListAttribute{
```

Comment:
I would rename to `security_group_ids`, to be more precise on what must be provided

Reply:
I renamed it in the schema